### PR TITLE
storage corrupt test refactoring

### PR DESCRIPTION
**Pull request by jamescfkung**
- obj_corrupt function support return TEEC_Result error code
- add error messages in obj_corrupt function
- support corrupt multiple bytes
- initial filename array variables to all zeros

Signed-off-by: James Kung <james.kung@linaro.org>

### CODE REVIEW

File: host/xtest/xtest_20000.c [hunk 1]
```diff
@@ -335,8 +344,11 @@ static int obj_corrupt(TEEC_UUID *p_uuid, void *file_id,
 		}
 
 		res = stat(name, &st);
-		if (res < 0)
+		if (res < 0) {
+			perror("Failed to get file information");
```
Comment (jenswi-linaro) [inline host/xtest/xtest_20000.c]: `fprintf(stderr, "stat(\"%s\"): %s", strerror(errno));` is preferable
Please apply the corresponding pattern for all the other cases with perror() below.

File: host/xtest/xtest_20000.c [hunk 2]
```diff
@@ -350,42 +362,63 @@ static int obj_corrupt(TEEC_UUID *p_uuid, void *file_id,
...
-		if (0 != fseek(fd, offset, SEEK_SET))
+		if (0 != fseek(fd, offset, SEEK_SET)) {
+			perror("Failed to seek file");
+			tee_res = TEEC_ERROR_BAD_PARAMETERS;
 			goto exit;
+		}
 
-		if (SIZE*NUMELEM != fread(&byte, SIZE, NUMELEM,fd))
+		res = fread(bytes, SIZE, NUMELEM, fd);
+		if (res != (int)(SIZE * NUMELEM)) {
```
Comment (jforissier) [inline host/xtest/xtest_20000.c]: Did you try with `(SIZE * NUMELEM) != 1`? It seems the tests with `CORRUPT_FILE_LAST_BYTE` will always fail.

File: host/xtest/xtest_20000.c (hunk 1 above)
Comment (jbech-linaro) [inline host/xtest/xtest_20000.c]: +1, better to separate stdout and stderr.

**Comment by jbech-linaro** [comment]
Besides comments from Jens and Jerome:
`Reviewed-by: Joakim Bech <joakim.bech@linaro.org>`

File: host/xtest/xtest_20000.c (hunk 1 above)
Comment (jamescfkung) [inline host/xtest/xtest_20000.c]: ok, will fix it.

File: host/xtest/xtest_20000.c (hunk 2 above)
Comment (jamescfkung) [inline host/xtest/xtest_20000.c]: I only try (SIZE * NUMELEM) != 1 when running xtest 20001. I did not verify with other test cases. Will fix it.

**Comment by jamescfkung** [comment]
- Fixed tests error when SIZE*NUMELEM > 1
- Replace perror with fprintf(stderr, ...)
- Code rebase

**Comment by jforissier** [comment]
`Tested-by: Jerome Forissier <jerome.forissier@linaro.org> (HiKey)`

**Comment by jamescfkung** [comment]
update tags in commit message

**Pull request merged**